include/engine/datafacade/shared_memory_datafacade.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef SHARED_MEMORY_DATAFACADE_HPP
+#define SHARED_MEMORY_DATAFACADE_HPP
+
+// implements all data storage when shared memory _IS_ used
+
+#include "storage/shared_barriers.hpp"
+#include "storage/shared_datatype.hpp"
+#include "storage/shared_memory.hpp"
+#include "engine/datafacade/bigramblock_datafacade_base.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+namespace datafacade
+{
+
+/**
+ * This datafacade uses an IPC shared memory block as the data location.
+ * Many SharedDataFacade objects can be created that point to the same shared
+ * memory block.
+ */
+class SharedDataFacade : public BigRAMBlockDataFacadeBase
+{
+
+  protected:
+    std::unique_ptr<storage::SharedMemory> m_layout_memory;
+    std::unique_ptr<storage::SharedMemory> m_large_memory;
+    std::shared_ptr<storage::SharedBarriers> shared_barriers;
+    storage::SharedDataType layout_region;
+    storage::SharedDataType data_region;
+    unsigned shared_timestamp;
+
+    SharedDataFacade() {}
+
+  public:
+    // this function handle the deallocation of the shared memory it we can prove it will not be
+    // used anymore
+    virtual ~SharedDataFacade()
```

Comment:
I think some functions below can throw which is a no-go here in the dtor.

Either try-catch them and do "something" about it, or add `noexcept(true)` to the dtor declaration which at least shuts down hard as soon as an exception is getting thrown.